File: test-samples/js-scenarios/scenario_safe.js
// Scenario: Quantum-Safe Cryptographic Usage
const crypto = require('crypto');
const bcrypt = require('bcrypt');
const argon2 = require('argon2');

// AES-256-GCM - QUANTUM RESISTANT
const key = crypto.randomBytes(32);
const iv = crypto.randomBytes(12);
const cipher = crypto.createCipheriv('aes-256-gcm', key, iv);
let encrypted = cipher.update('secret data', 'utf8', 'hex');
encrypted += cipher.final('hex');
const authTag = cipher.getAuthTag();

// AES-256-CBC - QUANTUM RESISTANT
const ivCBC = crypto.randomBytes(16);
const cipherCBC = crypto.createCipheriv('aes-256-cbc', key, ivCBC);

// ChaCha20-Poly1305 - QUANTUM RESISTANT
const chachaKey = crypto.randomBytes(32);
const chachaNonce = crypto.randomBytes(12);
const chachaCipher = crypto.createCipheriv('chacha20-poly1305', chachaKey, chachaNonce, { authTagLength: 16 });

// SHA-256/384/512 hashing - QUANTUM RESISTANT
const hash256 = crypto.createHash('sha256').update('data').digest('hex');
const hash384 = crypto.createHash('sha384').update('data').digest('hex');
const hash512 = crypto.createHash('sha512').update('data').digest('hex');
const hash3 = crypto.createHash('sha3-256').update('data').digest('hex');

// HMAC - QUANTUM RESISTANT
const hmac = crypto.createHmac('sha256', 'secret').update('data').digest('hex');

// bcrypt password hashing - QUANTUM RESISTANT
async function hashPassword() {
    const saltRounds = 12;
    const hash = await bcrypt.hash('password', saltRounds);
    const match = await bcrypt.compare('password', hash);
}

// Argon2 password hashing - QUANTUM RESISTANT
async function hashWithArgon2() {
    const hash = await argon2.hash('password');
    const valid = await argon2.verify(hash, 'password');
}

// PBKDF2 key derivation - QUANTUM RESISTANT
const derivedKey = crypto.pbkdf2Sync('password', 'salt', 100000, 32, 'sha256');

// scrypt key derivation - QUANTUM RESISTANT
const scryptKey = crypto.scryptSync('password', 'salt', 32);

File: test-samples/python-scenarios/scenario_safe.py
# Scenario: Quantum-Safe Cryptographic Usage
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.backends import default_backend
import os
import bcrypt
import argon2

# AES-256-GCM - QUANTUM RESISTANT
key = os.urandom(32)
iv = os.urandom(12)
cipher = Cipher(algorithms.AES(key), modes.GCM(iv), backend=default_backend())
encryptor = cipher.encryptor()
ciphertext = encryptor.update(b"secret data") + encryptor.finalize()

# ChaCha20-Poly1305 - QUANTUM RESISTANT
from cryptography.hazmat.primitives.ciphers.aead import ChaCha20Poly1305
chacha = ChaCha20Poly1305(os.urandom(32))
nonce = os.urandom(12)
ct = chacha.encrypt(nonce, b"secret", b"associated_data")

# SHA-256/SHA-384/SHA-512 - QUANTUM RESISTANT
digest = hashes.Hash(hashes.SHA256(), backend=default_backend())
digest.update(b"data")
hash_value = digest.finalize()

# bcrypt password hashing - QUANTUM RESISTANT
password_hash = bcrypt.hashpw(b"password", bcrypt.gensalt())

# Argon2 password hashing - QUANTUM RESISTANT
ph = argon2.PasswordHasher()
hash = ph.hash("password")

# HKDF key derivation - QUANTUM RESISTANT
from cryptography.hazmat.primitives.kdf.hkdf import HKDF
hkdf = HKDF(
    algorithm=hashes.SHA256(),
    length=32,
    salt=None,
    info=b"context",
    backend=default_backend()
)
derived_key = hkdf.derive(b"input key material")
